### ISSUE TITLE
Please order mapbox-gl-js API docs alphabetically

### ISSUE THREAD

**Issue by gmaclennan** [mapbox/mapbox-gl-js#4312]
I often find myself scrolling up and down the API docs searching for a particular method. It would be really helpful if the docs for the API methods on `Map` were organized alphabetically.

https://www.mapbox.com/mapbox-gl-js/api/#Map

**Comment by tmcw** [comment]
If we order them alphabetically, that will put `getZoom` near the beginning of the list and `setZoom` near the end. Will that still constitute an improvement in readability?

**Comment by gmaclennan** [comment]
Good point. At least the way I use the docs I will normally be looking for a specific method - like I need to get a particular value, like `getZoom`, and then am looking up and down the nav searching for it. `setZoom` being next to `getZoom` is only useful once you have found `getZoom`.

Perhaps one solution is to leave the main pane in the current order, but order the left-navigation alphabetically. Right now I can't figure out the logic behind the ordering so I find it hard to navigate. I almost always resort to command-F.

I'd be interested to know if other people have had the same experience though.

**Comment by stevage** [comment]
I have the same issue. Suggestion:

1. Break the method up on camel case boundaries.
2. If the first token is in [get, set, add, move, remove, has], sort by the rest of the word alphabetically, followed by the first token.
3. Otherwise sort by the whole word alphabetically.

Not perfect, but better. And the `zoomTo` edge case (is it best grouped with `getZoom` or `flyTo`?) will go away soon enough...

I'd suggest ordering alphabetically, but ignoring prefixed `get` or `set`

**Comment by jaapster** [comment]
Suggestion: add a sorting toggle

**Comment by colleenmcginnis** [comment]
I'm transferring this to the new home for the Mapbox GL JS documentation ([/mapbox-gl-js-docs](https://github.com/mapbox/mapbox-gl-js-docs)) since this would be handled in the table of contents component. See https://github.com/mapbox/mapbox-gl-js/pull/8582 for details on this move.
